javaslang/src-gen/main/java/javaslang/Tuple.java

```diff
@@ -95,6 +96,15 @@ static Tuple0 empty() {
     }
 
     /**
+    * Creates a tuple of two elements from java.util.Map.Entry.
+    * @param entry A java.util.Map.Entry
+    * @return a tuple of two elements.
+    */
+     static <T1, T2> Tuple2<T1, T2> fromEntry(Map.Entry<T1, T2> entry) {
```

Comment:
I think we can widen the argument types:

```java
static <T1, T2> Tuple2<T1, T2> fromEntry(Map.Entry<? extends T1, ? extends T2> entry) { ...
```

Only immutable types are covariant in generic arguments, here is the classical counter example (taken from [StackOverflow](http://stackoverflow.com/a/18666878/1110815)):

```java
// Illegal code - because otherwise life would be Bad
List<Dog> dogs = new List<Dog>();
List<Animal> animals = dogs; // Awooga awooga
animals.add(new Cat());
Dog dog = dogs.get(0); // This should be safe, right?
```

Projected to our use-case, this can't happen:

```java
// Illegal code - because otherwise life would be Bad
Map.Entry<T, Dog> t_dog = new SimpleEntry<>(t, dog);
Tuple<T, Animal> t_animal = Tuple.fromEntry(entry);
Tuple<T, Animal> t_animal2 = t_animal.update2(new Cat());
Dog dog = t_dog.getValue(); // ok
```

**Side-note:** We can improve the generics of other types, too. We will do it in another PR.

```java
public interface Validation<E, T> extends Value<T> {
    // should be `Either<? extends E, ? extends T> either`
    static <E, T> Validation<E, T> fromEither(Either<E, T> either) { ... }
}
```